shmat.c

```diff
@@ -0,0 +1,108 @@
+#include  <sys/types.h>
+#include  <sys/ipc.h>
+#include  <sys/shm.h>
+#include  <stdio.h>
+#include <stdlib.h>
+#include <getopt.h>
+
+int debug_print = 0;
+int i_from = 30;
+int i_till = 39;
+
+typedef struct node {
+	int value1;
+	void* value2;
```

Comment:
Reading "value1" and "value2" might be confusing.  It would be clearer to name them according to what they are representing (shmid, shm_ptr)